### PR TITLE
CPU refactor, sine cera

### PR DESCRIPTION
**Pull request by saagarjha**

### CODE REVIEW

**Comment by Rexogamer** [comment]
so the idea is to split things into separate prs? fair

**Comment by saagarjha** [comment]
Actually the idea is to finish the implementation this time

File: util/sync.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <pthread.h>
 #include <stdbool.h>
 #include <setjmp.h>
+#include <sys/errno.h>
```
Comment (tbodt) [inline util/sync.h:8]: Shouldn't this just be errno.h?

Comment (saagarjha) [reply util/sync.h:8]: Probably

**Pull request merged**